docs/development.md

```diff
@@ -0,0 +1,181 @@
+# Development
+
+## Structure
+
+This repository is a monorepo. Here's the list of all the apps and packages in it (each app has its own README):
+
+### Apps
+
+- [`apps/storefront`](../apps/storefront/README.md): Next.js e-commerce storefront starter pack
+- [`apps/saleor-app-checkout`](../apps/saleor-app-checkout/README.md): a Next.js Saleor app with a dashboard for managing settings and theme, backend for Checkout, and the Checkout Storefront – ready to be extended/modified
+
+### Packages
+
+- `packages/ui-kit`: UI kit for Storefront and Checkout
+- `packages/config-checkout`: `eslint` configuration
+- `packages/config-storefront`: `eslint` configuration
+- `packages/tsconfig`: `tsconfig.json`s used throughout the monorepo
+- `packages/checkout-storefront`: Checkout storefront component
+- `packages/checkout-common`: Common types and utils
+
+## Getting started
+
+### Prerequisites
+
+This monorepo uses `pnpm` as a package manager, so you will have to install it beforehand. To install it, run:
+
+```bash
+npm install -g pnpm
+```
+
+### Installing dependencies
+
+To install dependencies, run:
+
+```bash
+pnpm i
+```
+
+### Running
+
+To start all the dev servers, run:
+
+```bash
+pnpm dev
+```
+
+[The command will boot up 4 applications running on different ports.](#ports).
+
+You can also start a specific app by running:
+
+```bash
+pnpm dev --filter=saleor-app-checkout
+```
+
+#### [`saleor-app-checkout`](../apps/saleor-app-checkout/README.md)
+
+First, you need to change the default environment variables. Create `.env.local` file in `apps/saleor-app-checkout`, as described [here](../apps/saleor-app-checkout/README.md#env-variables).
+
+Then, to develop `saleor-app-checkout` app locally, you need to create a tunnel for it. The tunnel will enable you to display it within your Saleor Dashboard.
+
+If you want to read more about Saleor Apps architecture, see [the documentation](https://docs.saleor.io/docs/3.x/developer/extending/apps/key-concepts).
+
+To create a tunnel, run:
+
+```bash
+cd apps/saleor-app-checkout && npx saleor app tunnel 3001
+```
+
+> Note: The process needs to be running in the background. Please don't kill it during the development.
+
+<!-- is the bit below correct? you have to filter something to run it for all the apps? -->
+<!-- 
+To run the development server for all the apps, use the following command:
+
+```
+pnpm run dev --filter=saleor-app-checkout...
+```
+ -->
+#### [`checkout-storefront`](../packages/checkout-storefront/README.md)
+
+The URL structure for `checkout-storefront` is `localhost:3002?checkout=<ID>`, which means it requires a checkout id to work correctly. You don't have to supply it manually; it will be provided when transitioning from the cart to checkout.
```

Comment:
This bit is also outdated, we now support more than just `checkout` in query params.

Here's an example from staging-demo.saleor.io:

`?locale=pl-PL&checkout=Q2hlY2tvdXQ6ZTcwZjBlYzgtOWMyMS00Y2FkLWE4YzktZWQ3ODcwYjNhN2I5&channel=channel-pln`

The query parameters are:
- `locale` - locale, passed from storefront to checkout, will be used to display messages in correct language
- `checkout` - checkout id (nothing changed here)
- `channel` - channel slug for which we've created the checkout for

@bmigirl Correct me if I'm wrong:
Only `checkout` param is required

Reply:
All of these are required for prod, assuming we make storefronts just point to proper urls. For dev I'd say the assumption is that it's best to supply all for best performance, but we do have some hardcoded defaults just so the code doesn't break in case you forget. That's the best answer I can give

Reply:
I moved the instructions to `packages/checkout-storefront/README.md`, as suggested with `apps/saleor-app-checkout`.